qiskit_nature/second_q/mappers/bosonic_linear_mapper.py

```diff
@@ -0,0 +1,180 @@
+    :math:`|0_{n_k^{max}}, 0_{n_k^{max} - 1},..., 0_{n_k + 1}, 1_{n_k}, 0_{n_k - 1},..., 0_{0_k}\\rangle`
+
+    It implements the formula in Section II.C of Reference [1]:
+
+    .. math::
+        b_k^\\dagger = \\sum_{n_k =0}^{n_k^{max}-1}(\\sqrt{n_k +1}\\sigma_{n_k}^{+}\\sigma_{n_k + 1}^{-})
+
+    from :math:`n_k = 0` to :math:`n_k^{max} + 1` where :math:`n_k^{max}` is the truncation order
+    (defined by the user).
+    In the following implementation, we explicit the operators :math:`\\sigma^+` and :math:`\\sigma^-`
+    with the Pauli matrices:
+
+    .. math::
+        \\sigma_{n_k}^+ := S_j^+ = 0.5 * (X_j + i*Y_j)\\
+        \\sigma_{n_k}^- := S_j^- = 0.5 * (X_j - i*Y_j)
+
+    The length of the qubit register is: ``BosonicOp.num_modes * (BosonicLinearMapper.truncation + 1)``
+
+    To use this mapper one can for example:
+
+    .. code-block:: python
+
+      from qiskit_nature.second_q.mappers import BosonicLinearMapper
+      from qiskit_nature.second_q.operators import BosonicOp
+      mapper = BosonicLinearMapper(truncation=1)
+      qubit_op = mapper.map(BosonicOp({'+_0 -_0': 1}, num_modes=1))
+
+    .. note::
+        Since this mapper uses a truncation in the representation of the Bosonic operators, the
+        commutation relation after the mapping differ from the standard ones. Please refer to
+        Section 4, equation 22 of Reference [2] for more details
+
+    References:
+        [1] A. Miessen et al., Quantum algorithms for quantum dynamics: A performance study on the
+        spin-boson model, Phys. Rev. Research 3, 043212.
+        https://link.aps.org/doi/10.1103/PhysRevResearch.3.043212
+        [2] R. Somma et al., Quantum Simulations of Physics Problems, Arxiv
+        https://doi.org/10.48550/arXiv.quant-ph/0304063
+
+    """
+
+    def _map_single(
+        self, second_q_op: BosonicOp, *, register_length: int | None = None
+    ) -> SparsePauliOp:
+        """Maps a :class:`~qiskit_nature.second_q.operators.SparseLabelOp` to a``SparsePauliOp``.
+
+        Args:
+            second_q_op: the ``SparseLabelOp`` to be mapped.
+            register_length: when provided, this will be used to overwrite the ``register_length``
+                attribute of the operator being mapped. This is possible because the
+                ``register_length`` is considered a lower bound in a ``SparseLabelOp``.
+
+        Returns:
+            The qubit operator corresponding to the problem-Hamiltonian in the qubit space.
+
+        Raises:
+            NotImplementedError: when `qiskit_nature.settings.use_pauli_sum_op` is set to `True`.
+            This value is deprecated and, thus, not supported by this new implementation.
+            Set it to `False` in order to use this mapper.
+        """
+        if settings.use_pauli_sum_op:
+            raise NotImplementedError(
+                "Setting not supported for this class. Please disable settings.use_pauli_sum_op."
+            )
```

Comment:
Since this is new code I am wondering if it might just be better to ignore the setting (perhaps log a warning if we feel that a use might try and switch this to using PauliSumOp). The BosonicMapper already states in the map(), which is overriding that of the qubitmapper, that only SparsePauliOp types are returned.

It was really done to allow someone with existing code to transition over to the SparsePauliOp with the existing code in Nature so on new code... though this. I am expecting that we would be changing the default value over to False as well anyway in the next release, before it gets removed entirely as an option later - more as a comment to  @mrossinek

Reply:
I explicitly decided against ignoring this in view of #795 being planned after this. When we silently just return a `SparsePauliOp` while the rest of the stack might still expect a `PauliSumOp`, this can lead to some odd edge cases.
Thus the decision to raise an error here explicitly.

Reply:
Ok, with combining this with an existing operator to form a MixedOperator I can see that. For the next release, as I mentioned, I imagine we would be changing the default value for that setting, such by default things do not fail here if you simply use this alone. We might want to change this text `Please disable settings.use_pauli_sum_op` to be a bit clearer what disable means, ie simply state it must be set False.

Reply:
Yes we can be more explicit in that warning. Maybe something along the lines of:
```
This class does not support the deprecated `settings.use_pauli_sum_op` value. Please set `settings.use_pauli_sum_op = False` in order to use this mapper.
```

Reply:
I changed the error message to the one suggested by @mrossinek